virtualizarr/tests/test_manifests/test_array.py

```diff
@@ -212,115 +220,83 @@ def test_broadcast_empty(self, shape, chunks, grid_shape, target_shape):
-            filters=None,
-            order="C",
-            shape=(5, 1, 20),
-            zarr_format=2,
-        )
-
+    def test_concat(self, array_v3_metadata):
+        # both manifest arrays in this example have the same metadata properties
         chunks_dict1 = {
-            "0.0.0": {"path": "/foo.nc", "offset": 100, "length": 100},
-            "0.0.1": {"path": "/foo.nc", "offset": 200, "length": 100},
+            "0.0.0": {"path": "/foo1.nc", "offset": 100, "length": 100},
         }
-        manifest1 = ChunkManifest(entries=chunks_dict1)
-        marr1 = ManifestArray(zarray=zarray, chunkmanifest=manifest1)
-
         chunks_dict2 = {
-            "0.0.0": {"path": "/foo.nc", "offset": 300, "length": 100},
-            "0.0.1": {"path": "/foo.nc", "offset": 400, "length": 100},
+            "0.0.0": {"path": "/foo2.nc", "offset": 200, "length": 100},
         }
+        manifest1 = ChunkManifest(entries=chunks_dict1)
         manifest2 = ChunkManifest(entries=chunks_dict2)
-        marr2 = ManifestArray(zarray=zarray, chunkmanifest=manifest2)
+        chunks = (5, 1, 10)
+        shape = (5, 2, 20)
+        metadata = array_v3_metadata(shape=shape, chunks=chunks)
 
-        result = np.concatenate([marr1, marr2], axis=1)
+        marr1 = ManifestArray(metadata=metadata, chunkmanifest=manifest1)
+        marr2 = ManifestArray(metadata=metadata, chunkmanifest=manifest2)
 
-        assert result.shape == (5, 2, 20)
-        assert result.chunks == (5, 1, 10)
-        assert result.manifest.dict() == {
-            "0.0.0": {"path": "file:///foo.nc", "offset": 100, "length": 100},
-            "0.0.1": {"path": "file:///foo.nc", "offset": 200, "length": 100},
-            "0.1.0": {"path": "file:///foo.nc", "offset": 300, "length": 100},
-            "0.1.1": {"path": "file:///foo.nc", "offset": 400, "length": 100},
-        }
-        assert result.zarray.compressor == zarray.compressor
-        assert result.zarray.filters == zarray.filters
-        assert result.zarray.fill_value == zarray.fill_value
-        assert result.zarray.order == zarray.order
-        assert result.zarray.zarr_format == zarray.zarr_format
+        # Concatenate along the first axis
+        concatenated = np.concatenate([marr1, marr2], axis=0)
+        assert concatenated.shape == (10, 2, 20)
+        assert concatenated.dtype == np.dtype("int32")
 
-    def test_concat_empty(self):
-        # both manifest arrays in this example have the same zarray properties
-        zarray = ZArray(
-            chunks=(5, 1, 10),
-            compressor={"id": "zlib", "level": 1},
-            dtype=np.dtype("int32"),
-            fill_value=0.0,
-            filters=None,
-            order="C",
-            shape=(5, 1, 20),
-            zarr_format=2,
+    # FAILING: TypeError: no implementation found for 'numpy.concatenate' on types that implement __array_function__: [<class 'virtualizarr.manifests.array.ManifestArray'>, <class 'numpy.ndarray'>]
```

Comment:
This is a bad bug and shouldn't be forgotten - it implies that somehow inside the manifest / `ManifestArray` concatenation implementation some array is the wrong type.

(This can't be being thrown from the top-level `np.concatenate` you can see in the test, so it must be being thrown by the lower-level `np.concatenate` call that is actually used to merge the manifests internally in `.array_api`.)

Reply:
Is there an issue for this so it is not forgotten? This seems unrelated to this PR, right?

Reply:
There is now! #433.

Reply:
@abarciauskas-bgse this test passes though doesn't it? It's not xfailed... So is there actually a bug here to fix?

Reply:
My apologies @TomNicholas for the confusion: I didn't recall writing this comment so I thought it was unrelated to this PR. I think we can close #433 as this must have been a failure that occurred in the process of this refactor.